### PR TITLE
gh-126206: make clinic now forcefully regenerates clinic code

### LINKED ISSUE #126206: `make regen-all` doesn't pass `-f` to argument clinic
# Bug report

### Bug description:

Output:

```none
python3.13 ./Tools/clinic/clinic.py --make --exclude Lib/test/clinic.test.c --srcdir .
Error in file 'Modules/posixmodule.c' on line 13087:
Checksum mismatch! Expected '9756767bdbdabe94', computed '1da32369b7d9fd4b'. Suggested fix: remove all generated code including the end marker, or use the '-f' option.
make: *** [clinic] Error 1
```

This makes it harder to fix merge conflicts, see: https://discuss.python.org/t/reducing-merge-conflicts-in-generated-code/68537/8

### CPython versions tested on:

CPython main branch

### Operating systems tested on:

macOS

### Linked PRs
* gh-126244

### PR DESCRIPTION
**Pull request by erlend-aasland**
Resolves #126206.

* Issue: gh-126206

### CODE REVIEW

**Review by corona10 — APPROVED** [review]
lgtm

File: Makefile.pre.in [hunk 1]
```diff
@@ -904,7 +904,7 @@ coverage-report: regen-token regen-frozen
 # Run "Argument Clinic" over all source files
 .PHONY: clinic
 clinic: check-clean-src
-	$(PYTHON_FOR_REGEN) $(srcdir)/Tools/clinic/clinic.py --make --exclude Lib/test/clinic.test.c --srcdir $(srcdir)
+	$(PYTHON_FOR_REGEN) $(srcdir)/Tools/clinic/clinic.py -f --make --exclude Lib/test/clinic.test.c --srcdir $(srcdir)
```
Comment (AA-Turner) [inline Makefile.pre.in]:
```suggestion
	$(PYTHON_FOR_REGEN) $(srcdir)/Tools/clinic/clinic.py --force --make --exclude Lib/test/clinic.test.c --srcdir $(srcdir)
```

File: Misc/NEWS.d/next/Build/2024-10-31-15-37-05.gh-issue-126206.oC6z2i.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+``make clinic`` now runs Argument Clinic using the ``-f`` option, thus
+forcefully regenerating generated code.
```
Comment (AA-Turner) [inline Misc/NEWS.d/next/Build/2024-10-31-15-37-05.gh-issue-126206.oC6z2i.rst]:
```suggestion
``make clinic`` now runs Argument Clinic using the ``--force`` option,
thus forcefully regenerating generated code.
```

**Review by AA-Turner — APPROVED** [review]
Minor preference for long options (more readable), but non-blocking.

A

**Comment by erlend-aasland** [comment]
Thanks for the reviews; I agree that using long options is preferred.

**Pull request merged**